src/state_machine.zig

```diff
@@ -366,12 +456,197 @@ pub const StateMachine = struct {
         }
     }
 
-    /// See the above comment for get_account().
+    /// See the comment for get_account().
     fn get_transfer(self: *StateMachine, id: u128) ?*Transfer {
         if (self.transfers.getEntry(id)) |entry| {
             return &entry.value;
         } else {
             return null;
         }
     }
+
+    /// See the comment for get_account().
+    fn get_commit(self: *StateMachine, id: u128) ?*Commit {
+        if (self.commits.getEntry(id)) |entry| {
+            return &entry.value;
```

Comment:
@ifreund just to double-check that this is safe, i.e. the way we capture the value not as a pointer and then return a pointer to `entry.value`? I think this should be safe, since `HashMap.getEntry()` already returns an optional pointer, i.e. we don't need to capture the optional payload as a pointer to a pointer.

Reply:
Yep, this is safe. The captured `entry` variable has type `*Entry` here which points into the hashmap's storage. Since we preallocate the hashmap's storage up front and always use the "assume capacity" function variants for insertion this pointer is stable.

Reply:
Great, thanks for the clear explanation.